functional/functional.go

```diff
@@ -0,0 +1,8 @@
+package functional
+
+type Producer[V any] func() V
+type ErrorableProducer[V any] func() (V, error)
```

Comment:
Is it standard practice to say `Errorable` as a prefix? I think in general I like `ProducerWithError`, for example, just for discoverability.

Reply:
Yeah, I didn't love that but didn't know if there was precedent for something better.